### DISCUSSION TITLE
Not able to set gateway group as default?

### DISCUSSION THREAD

**Discussion by leandrogbreve** [jaredhendrickson13/pfsense-api#557] [Q&A] [answered]
Hi everyone, 

It's my first time posting, and I wasn't able to find any related topics about this. So I apologize in advance if it's duplicated. 

I can set a gateway as default by making a PATCH request to /routing/gateway/default with a body like:

`{
    "defaultgw4": "WAN2GW",
    "apply": true
}`

However, if I send the name of an existing and valid gateway group (image below):

![image](https://github.com/user-attachments/assets/201c3285-a1b1-49e4-b8c2-d87055e4a7e0)

It returns: 

`
{
    "code": 404,
    "status": "not found",
    "response_id": "FOREIGN_MODEL_FIELD_VALUE_NOT_FOUND",
    "message": "Field 'defaultgw4' could not locate 'Routing Gateway' object with 'name' set to 'FAILOVER_LINK1_LINK2'",
    "data": []
}
`

I haven't seen in the docs anything regarding not being able to set a gateway group as default, so that is my question: isn't it possible to set a gateway group as default? 

Thanks!

Leandro.

**Comment by jaredhendrickson13** [comment, marked as answer]
Hmm yeah this is an oversight on my end, I missed this one when support for gateway groups was added. The DefaultGateway model is currently only looking for RoutingGateways when it should be looking for both RoutingGateways and RoutingGatewayGroups:
https://github.com/jaredhendrickson13/pfsense-api/blob/b62e4bae71d0cb403e902bbbc7071204c63218b8/pfSense-pkg-RESTAPI/files/usr/local/pkg/RESTAPI/Models/DefaultGateway.inc#L23-L33

I'll get this added in a patch, you can follow #558 for progress. Thanks for reporting!

**Reply by leandrogbreve** [reply]
I really appreciate the fast response. That's amazing.

Thanks @jaredhendrickson13 .
